docker/package/tezos_setup_wizard.py

```diff
@@ -232,6 +232,32 @@ def check_blockchain_data(self):
             return False
         return True
 
+    # Check <network>.xtz-shots.io/base.json and collect the most recent snapshot
+    # that is suited for the chosen history mode
+    def get_snapshot_link(self):
+        self.config["snapshot_url"] = None
+        network_url = f"https://{self.config['network']}.xtz-shots.io"
+        json_url = f"{network_url}/base.json"
+        try:
+            with urllib.request.urlopen(json_url) as url:
+                snapshot_array = json.load(url)
+        except (urllib.error.URLError, ValueError):
+            print(f"Couldn't collect snapshot metadata from {json_url}")
+            return
+
+        for artifact in snapshot_array:
+            if artifact["artifact_type"] != "tezos-snapshot":
+                continue
+
+            # We just need the first instance of the right snapshot. To speed filtering up,
+            # it is assumed metadata is sorted chronologically in descending order.
+            if artifact["history_mode"] == self.config["history_mode"] or (
+                self.config["history_mode"] == "archive"
+                and artifact["history_mode"] == "full"
+            ):
+                self.config["snapshot_url"] = artifact["url"]
+                return
```

Comment:
Not sure if it makes a difference in terms of performance (I suspect so tho) but how about using python's `filter` to remove the non-needed elements from the iterator and `next` to grab the first element from it?

I believe iterators are fairly efficient and should keep the memory down and avoid unnecessary computations.

Reply:
Also, I'm looking at the https://jakartanet.xtz-shots.io/base.json and it seems like the newest snapshot is actually the last one in the JSON, so seems like this snapshot array should be traversed from the tail

Reply:
is it ok? maybe it's time for another issue about `base.json` format?

Reply:
I think that there is no need for a separate issue, especially because this issue/MR is still waiting for https://github.com/oxheadalpha/tezos-k8s/issues/446